tests/scripts/check_names.py

```diff
@@ -509,18 +509,19 @@ def parse_identifiers(self, include, exclude=None):
                 previous_line = ""
 
                 for line_no, line in enumerate(header):
-                    # Skip parsing this line if a block comment ends on it,
-                    # but don't skip if it has just started -- there is a chance
-                    # it ends on the same line.
-                    if re.search(r"/\*", line):
-                        in_block_comment = not in_block_comment
-                    if re.search(r"\*/", line):
-                        in_block_comment = not in_block_comment
-                        continue
-
+                    # Terminate current comment?
                     if in_block_comment:
-                        previous_line = ""
-                        continue
+                        line = re.sub(r".*?\*/", r"", line, 1)
+                        in_block_comment = False
+                    # Remove full comments and string literals
+                    line = re.sub(r'/\*.*?\*/|(")(?:[^\\\"]|\\.)*"',
```

Comment:
Probably a matter of taste, but I would find it way more readable with two calls to `re.sub()`, one for comments and one for string literals.

(Btw, pre-existing, but while we're at it: we should probably ignore `//` comments as well.)

Reply:
It would be more readable but less correct: doing it all at once handles things like `"/*"` correctly. I should have explained this in a comment though, I'll add one.

Reply:
Good point, and thanks for adding a comment about that.